### PR TITLE
Edits to using_openshift/authorization as per trello card #522

### PR DESCRIPTION
**Pull request by bfallonf**
This is a pull request adding the information from https://trello.com/c/bl4hLxId/522-document-limit-access-to-project-resources-via-roles-manage-security-beta1

### CODE REVIEW

**Comment by bfallonf** [comment]
@deads2k A few questions:

1. I can't see what the commands in the end are doing exactly? I can see that some of them are using the example commands outlined above, but the others are using the <openshift> command. Is this because the actual commands are not finalized just yet?

2. I'm not sure i fully grasp the custom roles bit. Can the reader just make a combination of their favourite parts of already existing roles and create a new one?

3. In the Global Roles and Bindings bit, at the end of each list object you write "short curcuit on match". Would you be able to elaborate on this? Does it mean that the authorizing stops once it finds the correct place?

I think that's all. Thanks for the initial work on the section!

File: using_openshift/authorization.adoc [hunk 1]
```diff
@@ -10,78 +10,99 @@
...
+*Authorization is still in the experimental stage, so the API, command line, and rule expressions are subject to change or removal without notice.*
 
-Authorization happens as a separate step from authentication.  Authentication determines who (identity) is taking the action, but authorizations determines whether that identity is allowed to perform the action.
+Authorization determines whether an identity is allowed to perform any action. Note that authorization is a separate step from authentication, which is more about determining who (identity) is taking the action.
 
-Authorization is managed using rules, roles, and bindings.  A rule grants or denies the ability to perform an action.  A role is a collection of rules.  A binding associates a role with a list of users and groups.  Users and groups may be bound to multiple roles at the same time.
+Authorization is managed using rules, roles, and bindings. A rule determines who has the ability to perform an action, and a role is a collection of rules. Users and groups may be bound to multiple roles at the same time. Lastly, a binding associates a role with a list of users and groups.
```
Comment (deads2k) [inline using_openshift/authorization.adoc]: A rule doesn't determine who has the ability to perform an action.  The rule describes the permission, but it is the binding that decides who the rule is applied to.

**Comment by deads2k** [comment]
> I can't see what the commands in the end are doing exactly? I can see that some of them are using the example commands outlined above, but the others are using the command. Is this because the actual commands are not finalized just yet?

I interleaved the `describe` command to make it a little more obvious what was happening.  I think that `describe` will probably be how you look at these on the command line.



> I'm not sure i fully grasp the custom roles bit. Can the reader just make a combination of their favourite parts of already existing roles and create a new one?

Custom roles can be created, but we don't have any reasonable commands to make their creation easier.  We're also revisiting rule expression, so I decided to leave out any attempt to document creating a custom role beyond our intent for it to be possible.



> In the Global Roles and Bindings bit, at the end of each list object you write "short curcuit on match". Would you be able to elaborate on this? Does it mean that the authorizing stops once it finds the correct place?

Correct.  It means that no more rules are evaluated.  That bit is important since the rules bound to a user could conflict.  Only the first match in that hierarchy is respected at evaluation time.

File: using_openshift/authorization.adoc (hunk 1 above)
Comment (bfallonf) [inline using_openshift/authorization.adoc]: Changed to:

"A rule describes the permission that each user has in relation to performing an action, and a role is a collection of rules. Users and groups may be bound to multiple roles at the same time. Lastly, a binding associates a role with a list of users and groups, and decides which rules apply to who."

**Comment by bfallonf** [comment]
1. Ok. So it sounds like the two commands are doing the same thing. This is fine seeing as how these may change in the near future.

2. Sure, I might leave this alone too. Pretty much cos of the same reason.

3. Cool. I edited this a little. I tool out all the "short circuit on match" references, and put in a para:

"Once authorization finds the correct rule, no other proceeding rules are checked. This is because rules bound to a user could cause conflicts. Only the first match is considered upon evaluation."

@deads2k Is there anything else you'd like to make suggestions for? Otherwise I can put this to complete.

Thanks!

File: using_openshift/authorization.adoc [hunk 2]
```diff
@@ -10,78 +10,101 @@
...
 
-Authorization happens as a separate step from authentication.  Authentication determines who (identity) is taking the action, but authorizations determines whether that identity is allowed to perform the action.
+Authorization determines whether an identity is allowed to perform any action. Note that authorization is a separate step from authentication, which is more about determining who (identity) is taking the action.
 
-Authorization is managed using rules, roles, and bindings.  A rule grants or denies the ability to perform an action.  A role is a collection of rules.  A binding associates a role with a list of users and groups.  Users and groups may be bound to multiple roles at the same time.
+Authorization is managed using rules, roles, and bindings. A rule describes the permission that each user has in relation to performing an action, and a role is a collection of rules. Users and groups may be bound to multiple roles at the same time. Lastly, a binding associates a role with a list of users and groups, and decides which rules apply to which user.
```
Comment (deads2k) [inline using_openshift/authorization.adoc]: @liggitt Any thoughts on this paragraph?

Comment (liggitt) [inline using_openshift/authorization.adoc]: Suggestion:
"
Authorization is managed using rules, roles, and bindings:
* A rule describes a set of permitted verbs on a set of resource types.
* A role is a collection of rules.
* Bindings associate users and/or groups with a role."
"
We should avoid using "action" in multiple ways (below, it includes namespace)

**Comment by bfallonf** [comment]
@deads2k @liggitt Sure, I've made those adjustments. Thanks for the input. 

Is there anything else for this pull request?

Thanks!

**Comment by deads2k** [comment]
lgtm

File: _build_cfg.yml [hunk 3]
```diff
@@ -98,7 +98,9 @@ Topics:
   - Name: Deployments
     File: deployments
   - Name: Integrating external services
-    File: integrating_external_services  
+    File: integrating_external_services
+    Name: Authorization
```
Comment (adellape) [inline _build_cfg.yml]: Needs a `-` before the `Name:` here.

**Comment by bfallonf** [comment]
@deads2k Awesome. Thanks for the review!

@adellape I fixed the build file. This should be good to merge now.

**Pull request merged**